toolkit/tools/specreader/specreader.go

```diff
@@ -200,8 +244,40 @@ func createChroot(workerTar, buildDir, specsDir, srpmsDir string) (chroot *safec
 	return
 }
 
+func findSpecFiles(specsDir string, parseList map[string]bool) (specFiles []string, err error) {
+	// Find the filepath for each spec in the SPECS directory.
+	if len(parseList) == 0 {
+		specSearch, err := filepath.Abs(filepath.Join(specsDir, "**/*.spec"))
+		if err == nil {
```

Comment:
Using non-standard error handling patterns can make the code harder to read. If you want to attach a single error log to multiple calls, then you should place those multiple calls in a helper function.

That being said, I don't think `filepath.Abs` will ever actually fail (at least on a normal OS). So, you probably only need to attach the error log to the `filepath.Glob` call.

Reply:
This was a dirty copy/paste of a similar function to quickly get this going... agreed this function is a bit odd. I'll rework this to follow the more standard `if err != nil  {` pattern we use.